internal/scm/bitbucketcloud/bitbucket_cloud.go

```diff
@@ -0,0 +1,324 @@
+	return responsePRs, nil
+}
+
+func (bbc *BitbucketCloud) getPullRequests(ctx context.Context, repoName string) ([]pullRequest, error) {
+
+	var repoPRs []pullRequest
+	prs, _ := bbc.bbClient.Repositories.PullRequests.Gets(&bitbucket.PullRequestsOptions{Owner: bbc.workspaces[0], RepoSlug: repoName})
+	prBytes, _ := json.Marshal(prs)
+	bbPullRequests := &bitbucketPullRequests{}
+	err := json.Unmarshal(prBytes, bbPullRequests)
+	if err != nil {
+		return nil, err
+	}
+	for _, pr := range bbPullRequests.Values {
+		convertedPr, err := bbc.convertPullRequest(bbc.workspaces[0], repoName, &pr)
+		if err != nil {
+			return nil, err
+		}
+		repoPRs = append(repoPRs, convertedPr)
+	}
+	return repoPRs, nil
+}
+
+func (bbc *BitbucketCloud) convertPullRequest(project, repoName string, pr *bbPullRequest) (pullRequest, error) {
+	status, err := bbc.pullRequestStatus(pr)
+	if err != nil {
+		return pullRequest{}, err
+	}
+
+	return pullRequest{
+		repoName:   repoName,
+		project:    project,
+		branchName: pr.Source.Branch.Name,
+
+		prProject:  pr.Source.Repository.Project.Key,
+		prRepoName: pr.Source.Repository.Slug,
+		number:     pr.ID,
+
+		guiURL: pr.Links.Html.Href,
+		status: status,
+	}, nil
+}
+
+func (bbc *BitbucketCloud) pullRequestStatus(pr *bbPullRequest) (scm.PullRequestStatus, error) {
+	switch pr.State {
+	case stateMerged:
+		return scm.PullRequestStatusMerged, nil
+	case stateDeclined:
+		return scm.PullRequestStatusClosed, nil
+	}
+
+	return scm.PullRequestStatusSuccess, nil
+}
+
+func (bbc *BitbucketCloud) GetOpenPullRequest(ctx context.Context, repo scm.Repository, branchName string) (scm.PullRequest, error) {
+	repoFN := repo.FullName()
+	repoSlug := strings.Split(repoFN, "/")
+	repoPRs, _ := bbc.getPullRequests(ctx, repoSlug[1])
+	for _, repoPR := range repoPRs {
+		pr := pullRequest(repoPR)
+		if pr.branchName == branchName && pr.status == scm.PullRequestStatusSuccess {
+			return repoPR, nil
+		}
+	}
+	return nil, nil
+}
+
+func (bbc *BitbucketCloud) MergePullRequest(ctx context.Context, pr scm.PullRequest) error {
+	bbcPR := pr.(pullRequest)
+	repoSlug := strings.Split(bbcPR.guiURL, "/")
+	prOptions := &bitbucket.PullRequestsOptions{
+		ID:           fmt.Sprintf("%d", bbcPR.number),
+		SourceBranch: bbcPR.branchName,
+		RepoSlug:     repoSlug[4],
+		Owner:        bbc.workspaces[0],
+	}
+	_, err := bbc.bbClient.Repositories.PullRequests.Merge(prOptions)
+	return err
+}
+
+func (bbc *BitbucketCloud) ClosePullRequest(ctx context.Context, pr scm.PullRequest) error {
+	bbcPR := pr.(pullRequest)
+	repoSlug := strings.Split(bbcPR.guiURL, "/")
+	prOptions := &bitbucket.PullRequestsOptions{
+		ID:           fmt.Sprintf("%d", bbcPR.number),
+		SourceBranch: bbcPR.branchName,
+		RepoSlug:     repoSlug[4],
+		Owner:        bbc.workspaces[0],
+	}
+	_, err := bbc.bbClient.Repositories.PullRequests.Decline(prOptions)
+	return err
+}
+
+func (bbc *BitbucketCloud) GetRepositories(ctx context.Context) ([]scm.Repository, error) {
```

Comment:
This seems to only include repository name filtering (`--repo`), not for example per project which I think is by far the most popular one (organizations for GitHub).

And for specific repo names, it seems that all repositories are fetched. If possible, I think getting those repositories by name should be prefered.

Reply:
I don't think we had time to implement this change, is it ok if we keep this as is with repository filtering? Trying to avoid heavier refactoring of this PR as it has gotten large and would prefer to do it as a follow up